src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCheckedParameterTests.cs

```diff
@@ -1262,6 +1274,77 @@ .maxstack  1
     IL_0000:  ldarg.0
     IL_0001:  call       ""object..ctor()""
     IL_0006:  ret
+}");
+        }
```

Comment:
>} [](start = 8, length = 1)

Please test with missing `System.ArgumentException` type. Something like:
```C#
    var source =
@"class Program
{
    static void Main(string[] args!) { }
}";
    var comp = CreateCompilation(source);
    comp.MakeTypeMissing(WellKnownType.System_ArgumentNullException);
    comp.VerifyEmitDiagnostics(...);
{

```

Reply:
It only seems to work without MakeTypeMissing / MakeMemberMissing. How should I fix it?

Reply:
Perhaps call `SyntheticBoundNodeFactory.WellKnownMember()` in `ConstructIfStatementForParameter()` for the exception constructor, so that the appropriate diagnostic is generated.

---
In reply to: [301322050](https://github.com/dotnet/roslyn/pull/36979#discussion_r301322050) [](ancestors = 301322050)

Reply:
Added but the output appears to be the same. Will come back to it once the other low-hanging fruit are done.